modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -763,6 +765,63 @@ public abstract class Base${schemaName}ResourceImpl
 				throw new UnsupportedOperationException("This method needs to be implemented");
 			</#if>
 		}
+
+		<#if (privateParsersNeeded?size >= 1)>
+			<#if privateParsersNeeded?seq_contains("boolean")>
+				private Boolean _parseBoolean(String value){
+					Boolean booleanValue = null;
+					if (value != null){
+						booleanValue = Long.parseBoolean(value);
+					}
+
+					return booleanValue;
+				}
+			</#if>
+
+			<#if privateParsersNeeded?seq_contains("date")>
+				private java.util.Date _parseDate(String value){
+					java.util.Date dateValue = null;
+					if (value != null){
+						java.util.Date dateValue = new java.util.Date(value);
+					}
+					return dateValue;
+
+				}
+			</#if>
+
+			<#if privateParsersNeeded?seq_contains("double")>
+				private Double _parseDouble(String value){
+					Double doubleValue = null;
+						if (value != null){
+							doubleValue = Double.parseDouble(value);
```

Comment:
As boolean, Double, Integer and Long calls the parseX method of the X class, maybe we could generalize it instead of duplicating similar code

Reply:
I see your point. I'm going to try to resolve it that way

Reply:
Maybe I confused you with this comment, regarding the `_parseDate` method. Sorry!